### DISCUSSION TITLE
Links are all taking me to a 500 error

### DISCUSSION THREAD

**Discussion by chrisjameschamp** [JulianPrieber/llc-docker#57] [Issues]
I am guessing something is not setup correctly.

But when I click on a link it takes me to url.com/going/10?facebookurl

But I get a 500 Server Error

or more specifically in the logs...

HTTP/1.1" 500 6615

**Comment by JulianPrieber** [comment]
On docker?

**Comment by JulianPrieber** [comment]
Behind a reverse proxy or does this also happen when accessing directly over the ip and port?

**Comment by chrisjameschamp** [comment]
Yes it is running on docker, that's a good question let me see if it does it from IP:Port

**Comment by chrisjameschamp** [comment]
Same results if going IP:Port

**Comment by JulianPrieber** [comment]
Could you please turn on debugging in your config and try again?

**Comment by chrisjameschamp** [comment]
Now it wont let me login, I guess I have to restore from backup again, trying to make this persistent is becoming a real pain

**Comment by chrisjameschamp** [comment]
Is there a way to restore from backup?

**Comment by JulianPrieber** [comment]
> Is there a way to restore from backup?

Not really besides exctarcting the backup file inside the `/` (root) directory.

**Comment by chrisjameschamp** [comment]
Problem is with Unraid, I do not have easy access to the root directory of a container.  Ultimately I dont think Little Link Custom is going to work on Unraid, since there is no way to do persistent storage, I tried but apparently something goes wrong when it unraid backs up appdata.

**Comment by JulianPrieber** [comment]
Can you run docker commands from the Unraid shell?

Something like `docker pull julianprieber/littlelink-custom`

**Comment by chrisjameschamp** [comment]
I believe so

**Comment by JulianPrieber** [comment]
You can try to run `docker pull julianprieber/littlelink-custom` this wont break anything so if this works I could help you setting up your docker image with persitent storage.

**Comment by chrisjameschamp** [comment]
Yes that works fine, no issues

**Comment by JulianPrieber** [comment]
Alright, what port(s) would you like your docker to run on?

**Comment by chrisjameschamp** [comment]
HTTP Port 2880
HTTPS Port 28443

**Comment by JulianPrieber** [comment]
Please run the commands:

`docker volume create llc-unraid`

and

`docker run --detach --name littlelink-custom-unraid --publish 2880:80 --publish 28443:443 --restart unless-stopped --mount source=llc-unraid,target=/htdocs julianprieber/littlelink-custom`

Then your docker should be up and running without any issues.

**Comment by chrisjameschamp** [comment]
Where does it make the llc-unraid volume, is it in the docker img?

**Comment by JulianPrieber** [comment]
Its in one of your folders on your system. Everything inside that folder will persist even if you delete the docker container.

**Comment by chrisjameschamp** [comment]
Ok so according to the folks at unraid this will not work in unraid, because of the way shares are setup.  But they told me to do essentially what I did last night, which was create both the appdata folder, and the docker container without mapping anything.  Then copy the files from htdocs, to the appdata folder, rebuild mapping it in such a way.

However for whatever reason, overnight it created all the issues I now have with 404 and 500 errors galore.

They also recommended what would help on unraid, and apparently some other docker solutions; is to put the data that cannot be changed by the issue, maybe it's setup files, or whatever in htdocs, but all of the user data, the database, images, config files, in another folder.  For your normal installations it's just a seperate folder alongside htdocs, but allow in the setup for that folder to be mapped elsewhere, and then unraid users could map it to their appdata folder, and everything would work great.  If something were to happen to the container itself, they could simply reinstall and point that folder back to appdata and everything would be restored right away.

For now I am going to run it as non persistent and see if I run into any issues that way.  Fingers crossed

**Comment by chrisjameschamp** [comment]
Also trying to restore files from the backup.zip file causes a 500 error as well
